template/template.go

```diff
@@ -74,30 +76,76 @@ func Named(s string) (*named, error) {
 	return nil, errors.New("no matching template found")
 }
 
+var DefaultTemplate, _ = Parse("{{ .Prompt }}")
+
 type Template struct {
 	*template.Template
 	raw string
 }
 
-func (t *Template) String() string {
-	return t.raw
+// response is a template node that can be added to templates that don't already have one
+var response = parse.ActionNode{
+	NodeType: parse.NodeAction,
+	Pipe: &parse.PipeNode{
+		NodeType: parse.NodePipe,
+		Cmds: []*parse.CommandNode{
+			{
+				NodeType: parse.NodeCommand,
+				Args: []parse.Node{
+					&parse.FieldNode{
+						NodeType: parse.NodeField,
+						Ident:    []string{"Response"},
+					},
+				},
+			},
+		},
+	},
 }
 
-var DefaultTemplate, _ = Parse("{{ .Prompt }}")
+var funcs = template.FuncMap{
+	"toJson": func(v any) string {
+		b, err := json.Marshal(v)
+		if err != nil {
+			return ""
+		}
+
+		return string(b)
+	},
+	"add": func(a, b int) int {
```

Comment:
I think here we can just use existing built-ins (to avoid committing to helpers in the first pass) e.g.

```
{{ range $index, $item := range .Items }}
  {{ $last := eq (len (slice $.Items $index)) 1 }}
  is it last? {{ $last }}
{{ end }}
```

While not pretty, it will always work and we can "upgrade" it to a better last helper later